api/utils/keys/yubikey.go

```diff
@@ -46,38 +51,64 @@ const (
 
 var (
 	// We use slot 9a for Teleport Clients which require `private_key_policy: hardware_key`.
-	pivSlotNoTouch = piv.SlotAuthentication
+	PivSlotNoTouch = piv.SlotAuthentication
 	// We use slot 9c for Teleport Clients which require `private_key_policy: hardware_key_touch`.
 	// Private keys generated on this slot will use TouchPolicy=Cached.
-	pivSlotWithTouch = piv.SlotSignature
+	PivSlotWithTouch = piv.SlotSignature
 )
 
 // getOrGenerateYubiKeyPrivateKey connects to a connected yubiKey and gets a private key
 // matching the given touch requirement. This private key will either be newly generated
 // or previously generated by a Teleport client and reused.
 func getOrGenerateYubiKeyPrivateKey(touchRequired bool) (*PrivateKey, error) {
+	// TODO(Joerger): Pass ctx through method.
+	ctx := context.TODO()
+
 	// Use the first yubiKey we find.
-	y, err := findYubiKey(0)
+	y, err := FindYubiKey(0)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
 	// Get the correct PIV slot and Touch policy for the given touch requirement.
-	pivSlot := pivSlotNoTouch
+	pivSlot := PivSlotNoTouch
 	touchPolicy := piv.TouchPolicyNever
 	if touchRequired {
-		pivSlot = pivSlotWithTouch
+		pivSlot = PivSlotWithTouch
 		touchPolicy = piv.TouchPolicyCached
 	}
 
-	// First, check if there is already a private key set up by a Teleport Client.
+	// First, check if there is already a private key on the slot.
 	priv, err := y.getPrivateKey(pivSlot)
-	if trace.IsNotFound(err) {
+	switch {
+	case trace.IsCompareFailed(err):
+		// Found a certificate in the slot that was not generated by a Teleport client.
+		// Prompt the user to confirm before we overwrite the key and cert in this slot.
+		promptQuestion := fmt.Sprintf("%v\nWould you like to overwrite this slot's private key and certificate?", err)
```

Comment:
What does printing the `err` here show up as? Is it something we want to display to the user?

If not, we could instead do something like

```go
fmt.Sprintf("Found a certificate in the specified slot that was not generated by a Teleport client.\nWould you like to overwrite this slot's private key and certificate?"
```

Reply:
It is a user-readable error message. In another open PR there is a different compare failed error to display, so I'm going to leave it as is.

Reply:
FWIW I changed this logic in the other PR so it is more clear where the message is coming from and why db346884e3a0bd69c154ba8a0380d62174283471